subgraphs/_reference_/src/sdk/protocols/perpfutures/account.ts

```diff
@@ -0,0 +1,554 @@
+    if (dAct) return false;
+
+    dAct = new ActiveAccount(id);
+    dAct.save();
+
+    return true;
+  }
+
+  private trackActivity(activityType: ActivityType): void {
+    const days = getUnixDays(this.event.block);
+    const hours = getUnixHours(this.event.block);
+
+    const generalHourlyID = `${this.account.id.toHexString()}-hourly-${hours}`;
+    const generalDailyID = `${this.account.id.toHexString()}-daily-${days}`;
+
+    const generalActivity: AccountWasActive = {
+      daily: this.isActiveByActivityID(Bytes.fromUTF8(generalDailyID)),
+      hourly: this.isActiveByActivityID(Bytes.fromUTF8(generalHourlyID)),
+    };
+
+    const typeHourlyID = `${this.account.id.toHexString()}-hourly-${hours}-${activityType}`;
+    const typeDailyID = `${this.account.id.toHexString()}-daily-${days}-${activityType}`;
+
+    const typeActivity: AccountWasActive = {
+      daily: this.isActiveByActivityID(Bytes.fromUTF8(typeDailyID)),
+      hourly: this.isActiveByActivityID(Bytes.fromUTF8(typeHourlyID)),
+    };
+
+    if (activityType == ActivityType.DEPOSIT) {
+      this.protocol.addActiveDepositor(typeActivity);
+    } else if (activityType == ActivityType.BORROW) {
+      this.protocol.addActiveBorrower(typeActivity);
+    } else if (activityType == ActivityType.LIQUIDATOR) {
+      this.protocol.addActiveLiquidator(typeActivity);
+    } else if (activityType == ActivityType.LIQUIDATEE) {
+      this.protocol.addActiveLiquidatee(typeActivity);
+    }
+
+    this.protocol.addActiveUser(generalActivity);
+  }
+
+  getBytesId(): Bytes {
+    return this.account.id;
+  }
+
+  /**
+   *
+   * @param pool The pool where the liquidity was deposited.
+   * @param amounts The amount deposited of inputTokens.
+   * @param sharesMinted The amount of shares minted of outputToken.
+   * @param updateMetrics Optional, defaults to true. If true it will update the pool and protocol TVL and inputTokenBalance.
+   * @returns Deposit
+   */
+  deposit(
+    pool: Pool,
+    amounts: BigInt[],
+    sharesMinted: BigInt,
+    updateMetrics: bool = true
+  ): Deposit {
+    const depositId = this.getIdFromEvent(EventType.DEPOSIT);
+    const deposit = new Deposit(depositId);
+
+    deposit.hash = this.event.transaction.hash;
+    deposit.logIndex = this.event.logIndex.toI32();
+    deposit.protocol = this.protocol.getBytesID();
+    deposit.to = pool.getBytesID();
+    deposit.from = this.account.id;
+    deposit.account = this.account.id;
+    deposit.blockNumber = this.event.block.number;
+    deposit.timestamp = this.event.block.timestamp;
+    deposit.inputTokens = pool.getInputTokens();
+    deposit.outputToken = pool.getOutputToken();
+    deposit.inputTokenAmounts = amounts;
+    deposit.outputTokenAmount = sharesMinted;
+    deposit.amountUSD = this.getAmountUSD(pool.getInputTokens(), amounts);
+    deposit.pool = pool.getBytesID();
+    deposit.save();
+
+    if (updateMetrics) this.protocol.addTransaction(TransactionType.DEPOSIT);
```

Comment:
This doesn't match with what the comment of the `updateMetrics` param states. It says it will update the pool and protocol TVL values. We should either do that, or change the comment.

Same for withdraw, swap, borrow, etc

Reply:
Yeah, I will update the comment